src/gui/providers/ogr/qgsgeopackageitemguiprovider.cpp

```diff
@@ -148,11 +152,17 @@ void QgsGeoPackageItemGuiProvider::deleteGpkg()
 
       if ( !QFile::remove( path ) )
       {
-        QMessageBox::warning( nullptr, title, tr( "Could not delete GeoPackage." ) );
+        if ( context.messageBar() )
```

Comment:
it seems there's a repeated pattern if (messageBar) messageBar->pushMessage() else warning() or information(). Shouldn't that be factor for easier maintenance ?

Reply:
Ah, seems there's already a method for that. Fixed.